# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1483817432 18000
#      Sat Jan 07 14:30:32 2017 -0500
# Node ID a2fc803e9556113f03f1c25b88f1493e557283a5
# Parent  198fbbf6934abfd4164df307418b1f9d453140ad
* configure.ac: Use OCTAVE_CHECK_LIB to check for Xft library.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1810,26 +1810,24 @@ else
   ## Alias CPPFLAGS to CFLAGS.  This is closer to the true meaning
   ## of `pkg-config --cflags` output.
   FT2_CPPFLAGS="$FT2_CFLAGS"
   AC_SUBST(FT2_CPPFLAGS)
 fi
 
 ## Check for Xft library (when using X11)
 
-warn_xft=""
-if test "$have_x" = yes; then
-  PKG_CHECK_MODULES(XFT, [xft],
-    [AC_DEFINE(HAVE_XFT, 1, [Define to 1 if Xft is available.])],
-    [warn_xft="Xft library not found.  OpenGL graphics will not be fully functional."])
-fi
-
-if test -n "$warn_xft"; then
-  OCTAVE_CONFIGURE_WARNING([warn_xft])
-fi
+## FIXME: We don't use this library directly, so there's no need to
+## check for the header file.  Do we even need to check for the library?
+
+OCTAVE_CHECK_LIB(Xft, Xft,
+  [Xft library not found.  OpenGL graphics will not be fully functional.],
+  [], [XftFontOpen],
+  [], [don't use Xft library, OpenGL graphics will not be fully functional],
+  [], [xft])
 
 ## Check for fontconfig library
 
 warn_fontconfig=""
 PKG_CHECK_MODULES(FONTCONFIG, [fontconfig],
   [AC_DEFINE(HAVE_FONTCONFIG, 1, [Define to 1 if fontconfig is available.])],
   [warn_fontconfig="Fontconfig library not found.  OpenGL graphics will not be fully functional."])
 
